pkg/internal/apis/config/validate_test.go

```diff
@@ -243,3 +244,48 @@ func TestNodeValidate(t *testing.T) {
 		})
 	}
 }
+
+
+func TestPortValidate(t *testing.T) {
+	cases := []struct {
+		TestName     string
+		Port         int32
+		ExpectErrors error
+	}{
+		{
+			TestName:     "-1 port",
+			Port:         -1,
+			ExpectErrors: nil,
+		},
+		{
+			TestName:     "valid port",
+			Port:         10,
+			ExpectErrors: nil,
+		},
+		{
+			TestName:     "negative port",
+			Port:         -2,
+			ExpectErrors: errors.Errorf("invalid port number: -2"),
+		},
+		{
+			TestName:     "extra port",
+			Port:         65536,
+			ExpectErrors: errors.Errorf("invalid port number: 65536"),
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc //capture loop variable
+		t.Run(tc.TestName, func(t *testing.T) {
+			err := validatePort(tc.Port)
+			// the error can be:
+			// - nil, in which case we should expect no errors or fail
+			if err == nil && tc.ExpectErrors != nil {
+				t.Error("received no errors but expected errors for case")
+			}else if !reflect.DeepEqual(err, tc.ExpectErrors) {
```

Comment:
nit: should be `gofmt`ed (which would be `} else if ...`)

Reply:
`hack/update/gofmt.sh` should fix this

Reply:
Done. Thanks :)